src/compute_born.cpp

```diff
@@ -0,0 +1,806 @@
+{
+  invoked_array = update->ntimestep;
+
+  // zero out arrays for one sample
+
+  int i;
+  for (i = 0; i < nvalues; i++) values_local[i] = 0.0;
+
+  // Compute Born contribution
+  if (pairflag) compute_pairs();
+
+  // For now these functions are commented
+  if (bondflag) compute_bonds();
+
+  if (angleflag) compute_angles();
+
+  if (dihedflag) compute_dihedrals();
+
+  // Even if stated in Voyatzis-2012, improper and dihedrals
+  // are not exactly the same in lammps. Atoms order can depend
+  // on the forcefield/improper interaction used. As such,
+  // writing a general routine to compute improper contribution
+  // might be more tricky than expected.
+  // if (impflag) compute_impropers();
+
+  // sum Born contributions over all procs
+  MPI_Allreduce(values_local,values_global,nvalues,
+                MPI_DOUBLE,MPI_SUM,world);
+
+  int m;
+  for (m=0; m<nvalues; m++) {
+    vector[m] = values_global[m];
+  }
+}
+
+
+/*------------------------------------------------------------------------
+  compute Born contribution of local proc
+  -------------------------------------------------------------------------*/
+
+void ComputeBorn::compute_pairs()
+
+{
+  int i,j,m,ii,jj,inum,jnum,itype,jtype;
+  double delx,dely,delz;
+  double rsq,factor_coul,factor_lj;
+  double dupair,du2pair,rinv;
+  int *ilist,*jlist,*numneigh,**firstneigh;
+
+  int *type = atom->type;
+  int *mask = atom->mask;
+  int nlocal = atom->nlocal;
+  double *special_coul = force->special_coul;
+  double *special_lj = force->special_lj;
+  int newton_pair = force->newton_pair;
+
+  // invoke half neighbor list (will copy or build if necessary)
+  neighbor->build_one(list);
+
+  inum = list->inum;
+  ilist = list->ilist;
+  numneigh = list->numneigh;
+  firstneigh = list->firstneigh;
+
+  // loop over neighbors of my atoms
+
+  Pair *pair = force->pair;
+  double **cutsq = force->pair->cutsq;
+
+  // Declares born values
+
+  int a,b,c,d;
+  double xi[3];
+  double fi[3];
+  double xj[3];
+  double rij[3];
+  double pair_pref;
+  double r2inv;
+
+
+  m = 0;
+  while (m<nvalues) {
+      for (ii = 0; ii < inum; ii++) {
+        i = ilist[ii];
+        if (!(mask[i] & groupbit)) continue;
+
+        xi[0] = atom->x[i][0];
+        xi[1] = atom->x[i][1];
+        xi[2] = atom->x[i][2];
+        itype = type[i];
+        jlist = firstneigh[i];
+        jnum = numneigh[i];
+
+        for (jj = 0; jj < jnum; jj++) {
+          j = jlist[jj];
+          factor_lj = special_lj[sbmask(j)];
+          factor_coul = special_coul[sbmask(j)];
+          j &= NEIGHMASK;
+
+          if (!(mask[j] & groupbit)) continue;
+
+          xj[0] = atom->x[j][0];
+          xj[1] = atom->x[j][1];
+          xj[2] = atom->x[j][2];
+          delx = xi[0] - xj[0];
+          dely = xi[1] - xj[1];
+          delz = xi[2] - xj[2];
+          rij[0] = xj[0]-xi[0];
+          rij[1] = xj[1]-xi[1];
+          rij[2] = xj[2]-xi[2];
+          rsq = delx*delx + dely*dely + delz*delz;
```

Comment:
delx is just -rij[0], why not use the latter here? Because you compute the square, you don't need to worry about the sign, too.

Reply:
Yes I started using array after making the lower loop through the indices and didn't modify the code here. Thank you for your comments. I'll made the recommended modification before moving on to some test cases.